src/main/java/randoop/operation/TypedOperation.java

```diff
@@ -69,6 +84,111 @@ public abstract class TypedOperation implements Operation, Comparable<TypedOpera
     this.inputTypes = inputTypes;
     this.outputType = outputType;
     this.execSpec = execSpec;
+    if (operation.isMethodCall()) {
+      MethodCall methodCall = (MethodCall) operation;
+      Method m = methodCall.getMethod();
+      Class<?> declaringClass = m.getDeclaringClass();
+      String className = declaringClass.getName();
+      OperationSpecification OperationSpec = extractSpecificationFromMethod(m, className);
+      try (SequenceCompiler compiler = new SequenceCompiler()) {
+        // Use the createExecutableSpecification method
+        ExecutableSpecification annoSpec =
+            SpecificationTranslator.createExecutableSpecification(m, OperationSpec, compiler);
+        if (this.execSpec == null) {
+          this.execSpec = new ExecutableSpecification(); // Initialize to default value
+        }
+        this.execSpec = ExecutableSpecification.merge(annoSpec, this.execSpec);
+      } catch (Exception e) {
+        System.out.println("Exception occurred while creating and merging ExecutableSpecification");
+        e.printStackTrace();
+      }
+    }
+  }
+
+  /**
+   * Creates a specification for the given method based on Nonnull annotations for parameter and
+   * return types.
+   *
+   * @param method the method to extract the specification from
+   * @param className the name of the class that declares the method
+   * @return the specification for the method
+   */
+  private static OperationSpecification extractSpecificationFromMethod(
+      Method method, @ClassGetName String className) {
+    String methodName = method.getName();
+    List<String> parameterNames = getParameterNames(method);
+    List<@ClassGetName String> parameterTypes = getParameterTypes(method);
+
+    OperationSignature operation =
+        OperationSignature.forConstructorName(className, methodName, parameterTypes);
+    Identifiers identifiers = new Identifiers("receiver", parameterNames, "result");
+
+    List<Precondition> preconditions = new ArrayList<>();
+    List<Postcondition> postconditions = new ArrayList<>();
+    List<ThrowsCondition> throwsConditions = new ArrayList<>();
+
+    // Read parameter annotations
+    Parameter[] parameters = method.getParameters();
+    for (Parameter parameter : parameters) {
+      AnnotatedType annotatedType = parameter.getAnnotatedType();
+      for (Annotation annotation : annotatedType.getAnnotations()) {
+        if (annotation.annotationType().getSimpleName().equals("NonNull")) {
```

Comment:
Please also check for "NotNull".

Reply:
@mernst I changed the HashSets to LinkedHashSets. Let me know if there are any other changes that would be useful.